### ISSUE TITLE
Access pointer validation fails for nested HashMaps

### ISSUE THREAD

**Issue by pkwagner** [rkyv/rkyv#549]
Sorry to bug you with yet another bizarre HashMap problem, but I found another one on 0.8.1:

```rust
#[derive(Hash, PartialEq, Eq, rkyv::Archive, rkyv::Serialize, rkyv::Deserialize, Debug)]
#[rkyv(derive(Hash, PartialEq, Eq, Debug))]
struct Key(u8, u8);

fn main() {
    println!("Hello world!");

    let mut nested_map = std::collections::HashMap::new();
    nested_map.insert(1337u16, 42u16);

    let mut map = std::collections::HashMap::new();
    map.insert(Key(1, 2), nested_map.clone());
    map.insert(Key(3, 4), nested_map.clone());

    let encoded = rkyv::to_bytes::<rkyv::rancor::Error>(&map).unwrap();

    // This .unwrap() fails!
    let _decoded = rkyv::access::<rkyv::collections::swiss_table::ArchivedHashMap<ArchivedKey, rkyv::collections::swiss_table::ArchivedHashMap<u16, u16>>, rkyv::rancor::Error>(&encoded).unwrap();
}
```

The problem generally occurs when nesting two or more non-empty HashMaps. This time the execution does not halt, but the `check_pos_with_context` validation during `access` fails, reporting an `UnalignedPointer`. I tried to track the error a bit further, but unfortunately I'm not yet that familiar with the internal archive data structure rkyv uses and validates.

**Comment by djkoloski** [comment]
Thanks for reporting this issue. This was because the serialization order and validation order differed. 8cfe124f1a21bbd03910e6ccbe1010cdd9a9b8c0 fixes this.

**Comment by pkwagner** [comment]
Thanks!

**Comment by djkoloski** [comment]
Now released in 0.8.2
